src/MongoDB.Driver.Core/Core/Operations/RetryableWriteOperationExecutor.cs

```diff
@@ -135,6 +137,20 @@ private static bool AreRetryableWritesSupported(ConnectionDescription connection
                 connectionDescription.IsMasterResult.ServerType != ServerType.Standalone;
         }
 
+        private static bool ShouldRetryWrites<TResult>(IRetryableWriteOperation<TResult> operation, RetryableWriteContext context)
+        {
+            var writeConcern = operation.WriteConcern;
+            if (writeConcern != null && !writeConcern.IsAcknowledged)
+            {
+                return false;
+            }
+
+            return
+                context.RetryRequested &&
+                AreRetryableWritesSupported(context.Channel.ConnectionDescription) &&
+                !context.Binding.Session.IsInTransaction;
```

Comment:
This is the same logic as before (almost, it's the opposite truth value).

Reply:
Ack